member.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+type member struct {
+	memberID, title, forenames, surname, emailAddress string
+}
+
+func (m1 *member) equal(m2 *member) bool {
+	return m1.memberID == m2.memberID &&
```

Comment:
`return *m1 == *m2` would also work here.

You have to dereference the pointer (so that you're not comparing pointer values), but after that, a struct is identical if they share the same type and have the same values. Obviously if any member of the struct was also a pointer you'd have issues again though. So it really only works with simple types.

Reply:
Updated.